mantle/cmd/ore/gcloud/upload.go

```diff
@@ -127,60 +128,55 @@ func runUpload(cmd *cobra.Command, args []string) {
 	}
 
 	imageStorageURL := fmt.Sprintf("https://storage.googleapis.com/%v/%v", uploadBucket, imageNameGS)
+	if uploadWriteUrl != "" {
+		err = ioutil.WriteFile(uploadWriteUrl, []byte(imageStorageURL), 0644)
```

Comment:
This has us writing out the URL file before we know whether the upload was successful.

Reply:
yeah, the real question is - is that OK? This URL really just refers to the GS location, which is pretty much independent of if the image was created.

Should the user be using the written out file as an indicator of if the image create succeeded or should they be using the exit code from the program?

Reply:
> This has us writing out the URL file before we know whether the upload was successful.

clarification. At this point the image file upload to GS was successful, but we haven't attempted to create the image in Google Compue yet.

Reply:
> Should the user be using the written out file as an indicator of if the image create succeeded or should they be using the exit code from the program?

IMHO users should use the exit code to determine success. However, exit codes are ephemeral so I'd argue that writing out the location would lead a user to believe that they can get what they expect at the provided URL.

Reply:
> I'd argue that writing out the location would lead a user to believe that they can get what they expect at the provided URL.

They can get the file from that URL. The GS upload succeeded.